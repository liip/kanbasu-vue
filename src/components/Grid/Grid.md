```vue
<grid xAlign="center" yAlign="middle" :multiline="true">
  <grid-item :widths="{ default: '1/3' }">
    <box type="default" size="small">
      Grid Item
    </box>
  </grid-item>
  <grid-item :widths="{ default: '1/3', md: '1/4', lg: '1/5' }">
    <box type="default" size="small">
      Grid Item
    </box>
  </grid-item>
  <grid-item :widths="{ default: '1/3', lg: '1/5' }">
    <box type="default" size="small">
      Grid Item
    </box>
  </grid-item>
</grid>
```