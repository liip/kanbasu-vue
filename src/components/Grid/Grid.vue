<template>
  <div :class="classNames">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'grid',

  props: {
    /**
     * The Grid gutters size:
     * `tight, tiny, small, large, huge`
     */
    size: {
      type: String,
    },
    /**
     * Horizontal alignment of the columns:
     * `center, right`
     */
    xAlign: {
      type: String,
    },
    /**
     * Vertical alignment of the columns:
     * `middle, bottom, even`
     */
    yAlign: {
      type: String,
    },
    /**
     * Reverse columns order
     */
    rev: {
      type: Boolean,
    },
    /**
     * Add space between rows when there's multiple lines of grid items
     */
    multiline: {
      type: Boolean,
    },
  },

  computed: {
    classNames() {
      return [
        'grid',
        { [`grid--${this.size}`]: !!this.size },
        { [`grid--${this.xAlign}`]: !!this.xAlign },
        { [`grid--${this.yAlign}`]: !!this.yAlign },
        { 'grid--rev': this.rev },
        { 'grid--multiline': this.multiline },
      ];
    },
  },
};
</script>
