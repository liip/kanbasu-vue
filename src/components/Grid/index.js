import Grid from './Grid.vue';

export default Grid;
