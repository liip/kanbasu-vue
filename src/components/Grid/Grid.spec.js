import { shallowMount } from '@vue/test-utils';
import Grid from './Grid.vue';

describe('Grid.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(Grid);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with size', () => {
    const wrapper = shallowMount(Grid, {
      propsData: {
        size: 'small',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with x alignement', () => {
    const wrapper = shallowMount(Grid, {
      propsData: {
        xAlign: 'center',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with y alignement', () => {
    const wrapper = shallowMount(Grid, {
      propsData: {
        yAlign: 'middle',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders reversed', () => {
    const wrapper = shallowMount(Grid, {
      propsData: {
        rev: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as multiline', () => {
    const wrapper = shallowMount(Grid, {
      propsData: {
        multiline: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
