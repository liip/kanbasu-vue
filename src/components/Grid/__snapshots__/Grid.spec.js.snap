// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid.vue renders 1`] = `<div class="grid"></div>`;

exports[`Grid.vue renders as multiline 1`] = `<div class="grid grid--multiline"></div>`;

exports[`Grid.vue renders reversed 1`] = `<div class="grid grid--rev"></div>`;

exports[`Grid.vue renders with size 1`] = `<div class="grid grid--small"></div>`;

exports[`Grid.vue renders with x alignement 1`] = `<div class="grid grid--center"></div>`;

exports[`Grid.vue renders with y alignement 1`] = `<div class="grid grid--middle"></div>`;
